cmd/podman/load.go

```diff
@@ -60,49 +63,49 @@ func loadCmd(c *cliconfig.LoadValues) error {
 	}
 	defer runtime.Shutdown(false)
 
-	input := c.Input
-	if runtime.Remote && len(input) == 0 {
-		return errors.New("the remote client requires you to load via -i and a tarball")
-	}
-	if len(input) == 0 {
-		input = "/dev/stdin"
-		c.Input = input
+	if len(c.Input) > 0 {
+		if err := parse.ValidateFileName(c.Input); err != nil {
+			return err
+		}
+	} else {
+		if terminal.IsTerminal(int(os.Stdin.Fd())) {
+			return errors.Errorf("cannot read from terminal. Use command-line redirection or the --input flag.")
+		}
+		outFile, err := ioutil.TempFile("/var/tmp", "podman")
+		if err != nil {
+			return errors.Errorf("error creating file %v", err)
+		}
+		defer os.Remove(outFile.Name())
+		defer outFile.Close()
 
-		fi, err := os.Stdin.Stat()
+		inFile, err := os.OpenFile("/dev/stdin", 0, 0666)
```

Comment:
For portability, can you read from os.Stdin rather than opening /dev/stdin?

Reply:
Actually that simplifies the code. Thanks @jwhonce.
Fixed

Reply:
A 2-For!